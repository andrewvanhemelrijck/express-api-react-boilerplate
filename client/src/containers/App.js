import React, { Component } from 'react';

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      res: "no response"
    };
  }

  componentDidMount() {
    fetch('api')
      .then((res) => {
        if (res.ok) return res.json();
        console.error('There was an error fetching data: /api')
      })
      .then((res) => {
        this.setState({ res: res })
      })
  }

  render() {
    return (
      <div>
        <h1>React is running!</h1>
        <h2>{ this.state.res.title }</h2>
      </div>
    );
  }
}

export default App;
