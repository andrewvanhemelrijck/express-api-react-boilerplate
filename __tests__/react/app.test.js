import React from 'react';
import App from '../../client/src/containers/App';
import renderer from 'react-test-renderer';

describe('Index', () => {
  test('App component renders correctly', () => {
    const tree = renderer
      .create(<App />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});