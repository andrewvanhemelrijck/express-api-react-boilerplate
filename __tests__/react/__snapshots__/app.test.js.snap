// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index App component renders correctly 1`] = `
<div>
  <h1>
    React is running!
  </h1>
  <h2 />
</div>
`;
